main/src/main/scala/sbt/Main.scala

```diff
@@ -884,7 +885,8 @@ object BuiltinCommands {
     val exchange = StandardMain.exchange
     val s1 = exchange run s0
     exchange publishEventMessage ConsolePromptEvent(s0)
-    val exec: Exec = exchange.blockUntilNextExec
+    val minGCInterval = Project.extract(s1).getOpt(Keys.minForcegcInterval).getOrElse(1.minute)
```

Comment:
We probably should use `GCUtil.defaultMinForcegcInterval` instead of `1.minute`.
We can also take this opportunity to bump the min interval.

Reply:
Good points. I re-pushed with a value of 10.minutes, but I more or less plucked that out of thin air.

Reply:
I also reworked it to defer the full GC until the user has been idle for at least one minute. It will also only force full GC at most once between tasks.